crates/proto/src/udp/udp_client_stream.rs

```diff
@@ -199,6 +242,10 @@ impl<S: Send + Unpin, MF: MessageFinalizer> Future for UdpClientConnect<S, MF> {
                 .name_server
                 .take()
                 .expect("UdpClientConnect invalid state: name_server"),
+            bind_addr: self
```

Comment:
Both this and the one above are probably unnecessary. I think `SocketAddr` is `Copy` and I'm just being pedantic with the `name_server` in regards to the take and semantic usage of this struct. What do you think about removing the outer Optional from both `name_server` and `bind_addr` fields?

Reply:
Done, I've removed both options.